test/app.listen.js

```diff
@@ -1,14 +1,74 @@
 'use strict'
 
+var net = require('net')
 var express = require('../')
+var http = require('http');
+var assert = require('assert');
 
 describe('app.listen()', function(){
+  function getOpenPort() {
+    return new Promise(function(resolve, reject){
+      var server = net.createServer();
+      server.once('error', reject);
+      server.listen(0, function(){
+        server.close(resolve.bind(null, server.address().port));
+      });
+    });
+  }
+
+  function makeGetRequest(port){
+    return new Promise(function(resolve, reject){
+      http
+        .get('http://localhost:' + port, function(res){
+          var data = ''
+          res.on('data', function(chunk){
+            data += chunk;
+          });
+
+          res.on('end', function(){
+            resolve(data);
+          });
+        })
+        .on('error', reject);
+    })
+  }
```

Comment:
Sent a whole request and assert with the response to make sure we are not on using other server that started in the meantime

I can remove it and just do:

```js
http
        .get('http://localhost:' + port, function(res){
			res.destroy()
			resolve()
		})
		.on('error', reject);
```

and it is enough as it's unlikely that a server will start at the open port in the meantime

WDYT?